src/corelibs/U2Formats/src/BAMUtils.cpp

```diff
@@ -90,13 +90,21 @@ samfile_t *openSamWithFai(const QByteArray &samUrl, U2OpStatus &os) {
 
     QByteArray faiUrlData = faiUrl.toLocal8Bit();
     void *aux = (void *)faiUrlData.constData();
-    return samopen(samUrl.constData(), "r", aux);
+    return samopen(samUrl.getURLStringAnsi(), "r", aux);
```

Comment:
Memory leak.
I suggest to rework getURLStringAnsi() signature to return QByteArray

Reply:
Thanks! Good idea.